src/aleph/http.clj

```diff
@@ -150,7 +157,7 @@
 
    Param key                   | Description
    | ---                       | ---
-   | `ssl-context`             | an `io.netty.handler.ssl.SslContext` object or a map of SSL context options (see `aleph.netty/ssl-client-context` for more details), only required if a custom context is required
+   | `ssl-context`             | an `io.netty.handler.ssl.SslContext` object or a map of SSL context options (see `aleph.netty/ssl-client-context` for more details), only required if a custom context is desired. When passing an `io.netty.handler.ssl.SslContext` object, it must have an ALPN config matching the `http-versions` option (see also `aleph.netty/application-protocol-config`).
```

Comment:
Can we create an example for this?
Because, as an end user, that's not something I'm expecting to have to do...

Reply:
Why not! Where would such an example go so that users will find it, though? :thinking:

Reply:
Somewhere in this namespace? [1]
To be completely transparent with you, I'm not even sure what this comment is asking me to do.

[1] : https://github.com/clj-commons/aleph/blob/master/examples/src/aleph/examples/http.clj

Reply:
Ah nice, that fits the bill! And in fact, the example there is broken due to the same issue at the moment :sweat_smile:  We should probably also re-test all the examples as part of the release preparation process...

> To be completely transparent with you, I'm not even sure what this comment is asking me to do.

Hmm well I've been assuming that a custom `SslContext` will only be provided by advanced users who should be aware of ALPN and how it relates to HTTP/2. It should be very rarely necessary anyway since we expose most (if not all?) SSL context settings via the options map, in which case you only need to specify which HTTP versions you care about (via the `http-versions` option) and Aleph takes care of ALPN for you then... So I would tend to leave it as-is and perhaps have the comment point to the example instead of `aleph.netty/application-protocol-config`? Thoughts?

Reply:
Heh with https://github.com/clj-commons/aleph/pull/730, the example *does* work without changes again, so it can't serve as an example anymore. [`aleph.examples.http2/http2-ssl-ctx`](https://github.com/clj-commons/aleph/blob/5caa1311db08c3f94c87871f2a57cb54e5987cdd/examples/src/aleph/examples/http2.clj#L33)  *could* work but it's a server SSL context so it wouldn't work in this particular case... I think I'll leave it as-is, advanced users should be able to figure out what to do based on the current docstrings. Otherwise we'll probably hear about it!